webapp/packages/plugin-codemirror6/src/useEditorAutocompletion.ts

```diff
@@ -0,0 +1,88 @@
+import type React from 'react';
+import { useLayoutEffect, useState } from 'react';
+import { css } from 'reshadow';
+
+import type { ComponentStyle } from '@cloudbeaver/core-theming';
+import { clsx, isSafari } from '@cloudbeaver/core-utils';
+import { autocompletion, startCompletion } from '@codemirror/autocomplete';
+import type { Extension, EditorState } from '@codemirror/state';
+import { keymap } from '@codemirror/view';
+
+import type { IEditorRef } from './IEditorRef';
+
+const styles = css`
+  :global(.cm-fix-position) {
+    margin-left: var(--cm-fix-position-left);
+    margin-top: var(--cm-fix-position-top);
+    z-index: 100;
+  }
+  :global(.cm-fixed-hack) {
```

Comment:
Please avoid using the word 'hack' in class (or variable) names. It is preferable to come up with a meaningful name and add a comment to provide clarity.

Reply:
fixed